File: WebAPI/Controllers/BeaconTypeController.cs
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using ReachUp;

namespace ReachUpWebAPI.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class BeaconTypeController : ControllerBase
    {

        [Authorize(Roles = "adm, dev")]
        [HttpGet("GetAll")]
        public async Task<IActionResult> GetAll()
        {
           return Ok(await new BeaconType().GetAll());
        }

        /*[Authorize(Roles = "dev")]
        [HttpGet]
        public async Task<IActionResult> Get(int id)
        {
           if (!string.IsNullOrWhiteSpace(id.ToString()))
               return Ok(await new BeaconType().Get(id));
           return BadRequest("Parameters are null");
        }*/

        [Authorize(Roles = "dev")]
        [HttpPost]
        public async Task<IActionResult> Post([FromBody] BeaconType beaconType)
        {
            if (beaconType != null)
                return Ok(await beaconType.Add());
            return BadRequest("Parameters are null");
        }

        [Authorize(Roles = "dev")]
        [HttpPatch]
        public async Task<IActionResult> Patch([FromBody] BeaconType beaconType)
        {
            if (beaconType != null)
                return Ok(await beaconType.Update());
            return BadRequest("Parameters are null");
        }

        [Authorize(Roles = "dev")]
        [HttpDelete]
        public async Task<IActionResult> Delete(int id)
        {
            if (!string.IsNullOrWhiteSpace(id.ToString()))
                return Ok(await new BeaconType().Delete(id));
            return BadRequest("Parameters are null");
        }
    }
}

File: MapMaker/src/Controllers/BeaconTypeController.ts
import { BeaconType } from "../Models/BeaconType";
import { ApiConnection } from "../services/ApiConnection";

export module BeaconTypeController {
  export class clsBeaconTypeController extends ApiConnection.clsApiConnection {
    constructor(){
      super();
    }

    public async Add(id : number, name : string)
    {
       return await this.httpPost(`BeaconType/Post`, 
         new BeaconType.clsBeaconType(id, name)
         );
    }

    public async Update(id : number, name : string)
    {
       return await this.httpPatch(`BeaconType/Patch`, 
         new BeaconType.clsBeaconType(id, name)
         );
    }

    public async Delete(id : number)
    {
       return await this.httpDelete(`BeaconType/Delete?id=${id}`)
    }

    public async GetAll()
    {
       return await this.httpGet('BeaconType/GetAll');
    }

    public async Get(id : number)
    {
       return await this.httpGet(`BeaconType/Get?id=${id}`);
    }
  }
}